src/commands/new.rs

```diff
@@ -178,17 +178,30 @@ impl FromStr for ParameterValue {
     }
 }
 
-async fn values_from_file(file: impl AsRef<Path>) -> Result<HashMap<String, String>> {
-    let file = absolutize(file)?;
 
-    let mut buf = vec![];
-    File::open(&file)
-        .await?
-        .read_to_end(&mut buf)
-        .await
+/// This function reads a file and parses it as TOML, then
+/// returns the resulting hashmap of key-value pairs.
+async fn values_from_file(path: impl AsRef<Path>) -> Result<HashMap<String, String>> {
+    // Get the absolute path of the file we're reading.
+    let path = absolutize(path)
+        .with_context(|| anyhow!("Cannot find path to file"))?;
+
+    // Open the file.
```

Comment:
We could replace this stanza (down to line 200) with

```
let text = tokio::fs::read_to_string(&path)
    .await
    .with_context(|| format!("Failed to read text from values file {}", path.display()))?;
```

I think?

Reply:
Nice catch! Works fine here.